editor.planx.uk/src/@planx/components/List/Public/Context.tsx

```diff
@@ -24,50 +37,109 @@ export const ListProvider: React.FC<ListProviderProps> = ({
   children,
   schema,
 }) => {
-  const [activeIndex, setActiveIndex] = useState<number | undefined>(0);
-  const [userData, setUserData] = useState<UserData>(
-    schema.min === 0 ? [] : [generateNewItem(schema)],
-  );
+  const [activeIndex, setActiveIndex] = useState<number>(0);
```

Comment:
Making this a `number` only, and not `number | undefined` makes it much easier to work with. When unset, it gets switched to `-1` which mirrors how indexes are often handled (e.g. `Array.prototype.findIndex()`)